### PR TITLE
core: fix issue#24660, slove error messages about `ValueError` when use model with history

### PR DESCRIPTION
**Pull request by CheneyYin**
- **Description:**
This PR will slove error messages about `ValueError` when use model with history. 
Detail in #24660.
#22933 causes that `langchain_core.runnables.history.RunnableWithMessageHistory._get_output_messages`  miss type check of `output_val` if `output_val` is `False`. After running `RunnableWithMessageHistory._is_not_async`, `output` is `False`.
https://github.com/langchain-ai/langchain/blob/249945a572a58a307cab1ae79e199e85b2707ff0/libs/core/langchain_core/runnables/history.py#L323-L334
https://github.com/langchain-ai/langchain/blob/15a36dd0a2bce0ec07090350c39de2fbaf3f05bc/libs/core/langchain_core/runnables/history.py#L461-L471
~~I suggest that `_get_output_messages` return empty list when `output_val == False`.~~

- **Issue**:
  - #24660

- **Dependencies:**: No Change.

### CODE REVIEW

File: libs/core/langchain_core/runnables/history.py [hunk 1]
```diff
@@ -467,6 +467,8 @@ def _get_output_messages(
             return [output_val]
         elif isinstance(output_val, (list, tuple)):
             return list(output_val)
+        elif not output_val:
```
[the remark this answers is not in the document]
Comment (CheneyYin) [reply libs/core/langchain_core/runnables/history.py:470]: I will.

[the remark this answers is not in the document]
Comment (CheneyYin) [reply libs/core/langchain_core/runnables/history.py]: @eyurtsev PTAL, I have add a test case.

**Comment by CheneyYin** [comment]
@eyurtsev PTAL

**Review by eyurtsev — APPROVED** [review]

**Comment by baskaryan** [comment]
thank you @CheneyYin! very important fix

**Pull request merged**